src/utils/hooks/use-validation-config.ts

```diff
@@ -0,0 +1,37 @@
+import { useContext, useEffect, useState } from "react";
```

Comment:
created `useValidationConfig` to handle validation config only, the validation schema logic is not part of this hook. this hook is used by all the fields only

the old `useValidationSchema` is updated to store the generated validation schema state only, it is used by the `FrontendEngine` component. this reduces memory usage slightly as the `useEffect` and state updates are not used in the fields anymore.